### PR TITLE
Refactor general warnings

### PR DESCRIPTION
**Pull request by gabegma**
Part of  #383 

## Description:
* Similar refactoring to the general warnings, to eventually allow this module to only run on one dataset split.
* I further refactored syntactic warnings a bit, to use `cls` instead of `label` throughout, for consistency.

## Checklist:

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

File: azimuth/modules/dataset_analysis/dataset_warnings.py [hunk 1]
```diff
@@ -41,94 +41,108 @@ def compute_on_dataset_split(self) -> List[DatasetWarningsResponse]:  # type: ig
...
-        alert_norm = np.logical_or(
-            divergence_norm > max_delta_representation, divergence_norm < -max_delta_representation
+        divergence_norm_per_cls = (
+            count_norm_per_cls_per_split[DatasetSplitName.eval]
+            - count_norm_per_cls_per_split[DatasetSplitName.train]
+        )
+        alert_norm_per_cls = np.logical_or(
+            divergence_norm_per_cls > max_delta_representation,
+            divergence_norm_per_cls < -max_delta_representation,
         )
```
Comment (JosephMarinier) [inline azimuth/modules/dataset_analysis/dataset_warnings.py:115]: Would that work?
```suggestion
        alert_norm_per_cls = np.abs(divergence_norm_per_cls) > max_delta_representation
```

File: azimuth/modules/dataset_analysis/dataset_warnings.py [hunk 2]
```diff
@@ -41,94 +41,108 @@ def compute_on_dataset_split(self) -> List[DatasetWarningsResponse]:  # type: ig
...
 
+        first_dm = next(iter(dm_dict.values()))
+        cls_names = first_dm.get_class_names(labels_only=True)
+        cls_indices = np.arange(first_dm.get_num_classes(labels_only=True))
```
Comment (JosephMarinier) [inline azimuth/modules/dataset_analysis/dataset_warnings.py:119]: Hey, do we need to create this `cls_indices`? Or can we enumerate `cls_names` for example?
```suggestion
```
```diff
-                    for i in cls_indices
+                    for i, cls_name in enumerate(cls_names)
```
```diff
-                        name=cls_names[i],
+                        name=cls_name,
```

File: azimuth/plots/dataset_warnings.py [hunk 3]
```diff
@@ -85,19 +73,17 @@ def nb_samples_plot(
...
+    common_args = dict(
+        xref="paper",
+        showarrow=False,
+        x=-0.037,
+        font=dict(color=Colors.Orange, size=AXIS_FONT_SIZE),
+    )
+    for split, alert in alert_per_cls_per_split.items():
+        for idx, cls in enumerate(order):
+            if alert[cls]:
+                fig.add_annotation(text=alert_text_per_split[split], y=idx, **common_args)
```
Comment (JosephMarinier) [inline azimuth/plots/dataset_warnings.py:86]: Now that `common_args` is used in only one place (good job! 👍), feel free to integrate it directly, or not.
```suggestion
    for split, alert in alert_per_cls_per_split.items():
        for idx, cls in enumerate(order):
            if alert[cls]:
                fig.add_annotation(
                    text=alert_text_per_split[split],
                    xref="paper",
                    showarrow=False,
                    x=-0.037,
                    y=idx,
                    font=dict(color=Colors.Orange, size=AXIS_FONT_SIZE),
                )
```

**Review by JosephMarinier — COMMENTED** [review]
Niiice! 😌 So satisfying! 👍
Minor comments, but that looks very good!

File: azimuth/plots/dataset_warnings.py (hunk 3 above)
Comment (gabegma) [reply azimuth/plots/dataset_warnings.py:86]: Right, I missed that one! Thanks!!

File: azimuth/modules/dataset_analysis/dataset_warnings.py (hunk 2 above)
Comment (gabegma) [reply azimuth/modules/dataset_analysis/dataset_warnings.py:119]: Right, it wasn't useful in this case!

File: azimuth/modules/dataset_analysis/dataset_warnings.py (hunk 1 above)
Comment (gabegma) [reply azimuth/modules/dataset_analysis/dataset_warnings.py:115]: Absolutely, good catch!

**Review by JosephMarinier — APPROVED** [review]

**Pull request merged**